leash.go

```diff
@@ -41,15 +43,139 @@ import (
 	"github.com/honeycombio/honeytail/tail"
 )
 
+func watchFiles(ctx context.Context, options GlobalOptions, tc tail.Config, fileCh chan<- tail.FileToWatch, errCh chan<- error) {
+	defer close(fileCh)
+
+	filenames := []string{}
+
+	for _, filePath := range tc.Paths {
+		if filePath == "-" {
+			filenames = append(filenames, filePath)
+		} else {
+			files, err := filepath.Glob(filePath)
+			if err != nil {
+				errCh <- err
+				return
+			}
+			files = tail.RemoveStateFiles(files, tc)
+			files = tail.RemoveFilteredPaths(files, tc.FilterPaths)
+			filenames = append(filenames, files...)
+		}
+	}
+	for _, path := range filenames {
+		fileCh <- tail.FileToWatch{
+			Name:     path,
+			ReadFrom: options.Tail.ReadFrom,
+		}
+	}
+
+	if len(filenames) == 0 {
+		errCh <- errors.New("No files to tail found after removing state files and filtered files.")
+	}
+
+	// Don't watch for new files if we've set --tail.stop. Close the file
+	// chan and be done with it.
+	if options.Tail.Stop {
+		return
+	}
+
+	// After getting the first static group of files to watch, poll
+	// any directories with globs for new files
+	filewatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting FS watcher: %s", err)
+		os.Exit(1)
+	}
+
+	for _, file := range options.Reqs.LogFiles {
+		// Only watch if we're globbing
+		if strings.Contains(file, "*") {
+			expandedFiles, err := filepath.Glob(file)
+			if err != nil {
+				logrus.WithFields(logrus.Fields{
+					"err": err,
+				}).Fatal("Error globbing files")
+			}
+
+			glob := file
+
+			for _, file := range expandedFiles {
+				dir := filepath.Dir(file)
+				logrus.WithFields(logrus.Fields{
+					"glob": glob,
+					"file": file,
+					"dir":  dir,
+				}).Debug("Watching directory for changes due to glob")
+				if err := filewatcher.Add(dir); err != nil {
+					logrus.WithFields(logrus.Fields{
+						"dir": dir,
+						"err": err,
+					}).Fatal("Error watching directory")
+				}
+			}
+		}
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case fsEvent := <-filewatcher.Events:
+			logrus.Warn(fsEvent)
+			if fsEvent.Op == fsnotify.Create {
```

Comment:
do we need to handle removals or does that get handled by the tailer detecting EOF/read error?